develop/migration_mt_django.rst

```diff
@@ -5,76 +5,185 @@
+
+
+1. Introducing the tenant column to models belonging to an account
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+**1.1 Introducting the column to models belonging to an account**
+
+In order to scale out a multi-tenant model, it's essential that you can locate all records that belong to an account quickly.
+The reason is that we want that an ORM call like
+
+.. code-block:: python
+
+  Project.objects.filter(account_id=1).prefetch_related('tasks')
+
+Generating the following queries
+
+
+.. code-block:: postgresql
+
+  SELECT *
+  FROM "myapp_project" WHERE "myapp_project"."account_id" = 1;
+
+  SELECT *
+  FROM "myapp_task" WHERE "myapp_task"."project_id" IN (1, 2, 3);
+
+
+Can be changed to
+
+.. code-block:: postgresql
+
+  SELECT *
+  FROM "myapp_project" WHERE "myapp_project"."account_id" = 1;
+
+  SELECT *
+  FROM "myapp_task
+  WHERE ("myapp_task"."account_id" = 1 AND "myapp_task"."project_id" IN (1, 2, 3));
+
+
+This way you can easily query the tasks belonging to one account.
+The easiest way to achieve this is to simply add a :code:`account_id` column on every object that belongs to an account.
+
+In our case:
+
+.. code-block:: python
+
+  class Task(models.Model):
+      name = models.CharField(max_length=255)
+      project = models.ForeignKey(Project, on_delete=models.CASCADE,
+                                  related_name='tasks')
+      account = models.ForeignKey(Account, related_name='tasks',
+                                  on_delete=models.CASCADE)
+
+
+Create a migration to reflect the change: :code:`python manage.py makemigrations`.
+
+
+**1.2. Introduce a column for the account\_id on every ManyToMany model that belongs to an account**
 
-  class Product(models.Model):
-    name = models.CharField(max_length=255)
-    description = models.TextField()
-    price = models.DecimalField(max_digits=6, decimal_places=2),
-    quantity = models.IntegerField()
 
-    store = models.ForeignKey(Store, on_delete=models.CASCADE)
+The goal is the same as previously. We want to be able to have ORM calls and queries routed to one account. We also want to be able to distribute the ManyToMany relationship related to an account on the account_id.
 
-  class Purchase(models.Model):
-    ordered_at = models.DateTimeField(default=timezone.now)
-    billing_address = models.TextField()
-    shipping_address = models.TextField()
+So the calls generated by:
 
-    product = models.ForeignKey(Product, on_delete=models.CASCADE)
+.. code-block:: python
+
+  Project.objects.filter(account_id=1).prefetch_related('managers')
+
+
+Can include in their :code:`WHERE` clause the :code:`account_id` like this:
+
+.. code-block:: postgresql
+
+  SELECT *
+  FROM "myapp_project" WHERE "myapp_project"."account_id" = 1;
```

Comment:
I'll leave it up to you whether you prefer the query style generated by the ORM such as

```sql
SELECT *
FROM "myapp_project" WHERE "myapp_project"."account_id" = 1;
```

vs a more friendly style like

```sql
SELECT *
  FROM myapp_project
 WHERE account_id = 1;
```

The former has the advantage that it matches what's actually happening, but the latter I find easier to understand.